query-engine/core/src/executor/interpreting_executor.rs

```diff
@@ -78,21 +79,21 @@ where
         &self,
         tx_id: Option<TxId>,
         operations: Vec<Operation>,
-        transactional: bool,
+        transaction: Option<BatchDocumentTransaction>,
         query_schema: QuerySchemaRef,
         trace_id: Option<String>,
     ) -> crate::Result<Vec<crate::Result<ResponseData>>> {
         if let Some(tx_id) = tx_id {
             self.itx_manager.batch_execute(&tx_id, operations, trace_id).await
-        } else if transactional {
+        } else if let Some(transaction) = transaction {
             let connection_name = self.connector.name();
             let conn_span = info_span!(
                 "prisma:engine:connection",
                 user_facing = true,
                 "db.type" = connection_name.as_str()
             );
             let mut conn = self.connector.get_connection().instrument(conn_span).await?;
-            let mut tx = conn.start_transaction(None).await?;
+            let mut tx = conn.start_transaction(transaction.isolation_level).await?;
```

Comment:
I'm wondering if we should rather expose this as `transaction.isolation_level()` so making it a function.